libs/wingc/src/type_check.rs

```diff
@@ -74,21 +74,40 @@ pub enum SymbolKind {
 	Namespace(NamespaceRef),
 }
 
+#[derive(Debug, Clone)]
+pub enum VariableKind {
+	/// a free variable not associated with a specific type
+	Free,
```

Comment:
Is this meant to refer to any local variable? I think that might be better compiler nomenclature:
```suggestion
	/// a local variable not associated with a specific type
	Local,
```

Reply:
It's meant to represent a value that's not bound to an object.

Not sure local is the right term.

We've used "free variables" previously

Reply:
Sorry, I probably didn't explain myself well. I think we're thinking of the term "bound" in different ways (I'm using it in the math/Lisp sense where it refers to a property about substituting variables into functions, and are using it in a way that seems more related to association or membership to types). Anyway I'd be down to chat about it sometime, feels like a shared understanding of compiler terminology is useful long term for taming the complexity monster we have here 😁